src/validate.ts

```diff
@@ -209,29 +242,27 @@ export async function validatePrTitleBump(
 
   core.info(""); // for vertical whitespace
 
-  for (const commit of commits) {
-    try {
-      const cc = new ConventionalCommitMessage(commit);
-      highestBump = cc.bump > highestBump ? cc.bump : highestBump;
-    } catch (error) {
-      if (
-        !(
-          error instanceof ConventionalCommitError ||
-          error instanceof MergeCommitError ||
-          error instanceof FixupCommitError
-        )
-      ) {
-        throw error;
-      }
-      core.warning(`${baseError}, as the PR contains non-compliant commits`);
-      return false;
-    }
+  const results = processCommits(commits, config);
+
+  if (results.some(c => c.errors.length !== 0)) {
+    // Abort if the list contains any non-compliant commits; bump level
+    // validation only really makes sense if all commits are found to
+    // be compliant.
+    core.warning(`${baseError}, as the PR contains non-compliant commits`);
+    return false;
   }
 
+  const highestBump: SemVerType =
+    results.reduce((acc, val) => {
+      const accb = acc.message?.bump ?? SemVerType.NONE;
+      const valb = val.message?.bump ?? SemVerType.NONE;
+      return accb > valb ? acc : val;
+    }).message?.bump ?? SemVerType.NONE;
+
   if (highestBump !== prTitle.bump) {
-    const commitSubjects = commits.map(m => {
-      return m.split("\n")[0];
-    });
+    const commitSubjects = commits.map(
+      m => (m.message.match(/^.*$/m) ?? [""])[0]
```

Comment:
Duplication of logic bump.ts:153

Reply:
See: https://github.com/tomtom-international/commisery-action/pull/211#discussion_r1105441238